src/main/scala/com.snowplowanalytics.maxmind.iplookups/IpLookups.scala

```diff
@@ -14,113 +14,186 @@ package com.snowplowanalytics.maxmind.iplookups
 
 import java.io.File
 import java.net.InetAddress
-import java.util.{Collections, Map}
 
-import com.maxmind.db.CHMCache
-import com.maxmind.geoip2.model.CityResponse
-import com.maxmind.geoip2.DatabaseReader
-import com.snowplowanalytics.lrumap.LruMap
+import cats.{Eval, Id, Monad}
 import cats.effect.Sync
-import cats.syntax.either._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
+import cats.syntax.option._
+import com.maxmind.db.CHMCache
+import com.maxmind.geoip2.DatabaseReader
+import com.snowplowanalytics.lrumap.{CreateLruMap, LruMap}
 
 import model._
 
-/** Companion object to hold alternative constructors. */
-object IpLookups {
+sealed trait CreateIpLookups[F[_]] {
 
   /**
    * Create an IpLookups from Files
-   *
    * @param geoFile Geographic lookup database file
    * @param ispFile ISP lookup database file
    * @param domainFile Domain lookup database file
    * @param connectionTypeFile Connection type lookup database file
    * @param memCache Whether to use MaxMind's CHMCache
    * @param lruCacheSize Maximum size of LruMap cache
    */
-  def createFromFiles[F[_]: Sync](
+  def createFromFiles(
     geoFile: Option[File] = None,
     ispFile: Option[File] = None,
     domainFile: Option[File] = None,
     connectionTypeFile: Option[File] = None,
     memCache: Boolean = true,
     lruCacheSize: Int = 10000
-  ): F[IpLookups[F]] =
-    (
-      if (lruCacheSize > 0)
-        Sync[F].map(LruMap.create[F, String, IpLookupResult](lruCacheSize))(Some(_))
-      else Sync[F].pure(None)
-    ).flatMap((lruCache) =>
-      Sync[F].delay {
-        new IpLookups(
-          geoFile,
-          ispFile,
-          domainFile,
-          connectionTypeFile,
-          memCache,
-          lruCache
-        )
-    })
+  ): F[IpLookups[F]]
 
   /**
-   * Alternative constructor taking Strings rather than Files
-   *
+   * Alternative constructor taking filenames rather than files
    * @param geoFile Geographic lookup database filepath
    * @param ispFile ISP lookup database filepath
    * @param domainFile Domain lookup database filepath
    * @param connectionTypeFile Connection type lookup database filepath
    * @param memCache Whether to use MaxMind's CHMCache
    * @param lruCacheSize Maximum size of LruMap cache
    */
-  def createFromFilenames[F[_]: Sync](
+  def createFromFilenames(
     geoFile: Option[String] = None,
     ispFile: Option[String] = None,
     domainFile: Option[String] = None,
     connectionTypeFile: Option[String] = None,
     memCache: Boolean = true,
     lruCacheSize: Int = 10000
-  ): F[IpLookups[F]] =
-    IpLookups.createFromFiles(
-      geoFile.map(new File(_)),
-      ispFile.map(new File(_)),
-      domainFile.map(new File(_)),
-      connectionTypeFile.map(new File(_)),
-      memCache,
-      lruCacheSize
-    )
+  ): F[IpLookups[F]] = createFromFiles(
+    geoFile.map(new File(_)),
```

Comment:
In next version, we should standardize on `java.nio.file.Path` as `java.io.File` is mutable and unserializable.